### PR TITLE
Add uniform to onnx export

### LINKED ISSUE #42230: ONNX export does not support distributions.uniform
## 🐛 Bug

The operator torch.distributions.uniform is not supported by torch.onnx.export. It fails with the below message when a model with the uniform operation is exported: 

"RuntimeError: Exporting the operator uniform to ONNX opset version 12 is not supported. Please open a bug to request ONNX export support for the missing operator."

ONNX supports RandomUniform from opset 1, it looks to be a case of exporting the uniform operation to that. 

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

### PR DESCRIPTION
**Pull request by metalprogrammer**
Fixes #42230 and #54483
ONNX supports RandomUniform from opset 1 but there was not a file for said opset in torch/onxx, please advise if the symbol should be moved to a different file,

### CODE REVIEW

**Review by ezyang — CHANGES_REQUESTED** [review]
This isn't going to fly, renaming from/to is a BC-breaking change for traditional Python API. Why do you need to rename it anyway?

**Comment by ezyang** [comment]
Good point. Can you file an issue about this?

For your case, there is a simple workaround: accept the arguments as kwargs and then you can get out the from keyword argument.

cc @mruberry for possibility of making uniform accept low/high making it more similar to numpy

**Review by garymm — CHANGES_REQUESTED** [review]

File: test/onnx/test_operators.py [hunk 1]
```diff
@@ -584,6 +584,10 @@ def test_upsample_nearest_size(self):
         x = torch.randn(1, 2, 3, 4, requires_grad=True)
         self.assertONNX(lambda x: nn.functional.interpolate(x, size=16, mode="nearest"), x)
 
+    def test_uniform(self):
+        x = torch.tensor([1.0, 2.0])
+        self.assertONNX(lambda x: x.uniform_(2, to=2), x, operator_export_type=torch.onnx.OperatorExportTypes.ONNX_ATEN)
```
[the remark this answers is not in the document]
Comment (metalprogrammer) [reply test/onnx/test_operators.py:589]: @garymm I will remove this test. it looks like there are tests for uniform export in test/onnx/test_pytorch_onnx_onnxruntime.py so is there any further additions I should add there?
https://github.com/pytorch/pytorch/blob/master/test/onnx/test_pytorch_onnx_onnxruntime.py#L10030-L10064

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -3257,3 +3257,11 @@ def broadcast_tensors(g, self):
 
     t_list = [expand_as(g, t, t_with_final_shape) for t in all_tensors]
     return g.op("prim::ListConstruct", *t_list)
+
+
+@parse_args("v", "f", "f", "v")
+def uniform(g, self, from_, to, generator):
+    from_v = g.op("Constant", value_t=torch.tensor(from_, dtype=torch.float))
```
[the remark this answers is not in the document]
Comment (metalprogrammer) [reply torch/onnx/symbolic_opset9.py]: Previous version used them and just forgot to remove them, they are now gone

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -3257,3 +3257,11 @@ def broadcast_tensors(g, self):
...
+    to_v = g.op("Constant", value_t=torch.tensor(to, dtype=torch.float))
+    kwargs = {"from_f": from_, "to_f": to}
+    return g.op("Uniform", self, generator, **kwargs)
```
[the remark this answers is not in the document]
Comment (metalprogrammer) [reply torch/onnx/symbolic_opset9.py]: Completed

File: test/onnx/test_operators.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (garymm) [reply test/onnx/test_operators.py]: If there's no test that fails before this change and passes now then I don't understand what you're actually fixing or improving.
I just checked and exporting distributions.Uniform seems to work on the latest PyTorch nightly:

```python
!pip install --pre torch torchvision torchaudio -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html -U
import torch
print("torch version:", torch.__version__)

class M(torch.nn.Module):
    def forward(self, x, y):
        return torch.distributions.Uniform(x, y).sample().size(0), x , y

m = M()
inp = (torch.tensor([[0.0], [6.0]]), torch.tensor([[1.0], [7.0]]))
torch.onnx.export(
    m, inp,
    "/dev/null", verbose=True, opset_version=9)
```

Produces output:
```
torch version: 1.11.0.dev20211102+cpu
graph(%0 : Float(2, 1, strides=[1, 1], requires_grad=0, device=cpu),
      %1 : Float(2, 1, strides=[1, 1], requires_grad=0, device=cpu)):
  %2 : Long(2, strides=[1], device=cpu) = onnx::Shape(%0) # /usr/local/lib/python3.7/dist-packages/torch/functional.py:75:0
  %3 : Float(2, 1, device=cpu) = onnx::ConstantOfShape[value={0}](%2) # /usr/local/lib/python3.7/dist-packages/torch/functional.py:75:0
  %4 : Float(2, 1, strides=[1, 1], device=cpu) = onnx::Add(%3, %0) # /usr/local/lib/python3.7/dist-packages/torch/functional.py:75:0
  %5 : Float(2, 1, strides=[1, 1], device=cpu) = onnx::Add(%4, %1) # /usr/local/lib/python3.7/dist-packages/torch/functional.py:75:0
  %6 : Long(2, strides=[1], device=cpu) = onnx::Shape(%5) # /usr/local/lib/python3.7/dist-packages/torch/functional.py:75:0
  %7 : Float(2, 1, strides=[1, 1], requires_grad=0, device=cpu) = onnx::Expand(%0, %6) # /usr/local/lib/python3.7/dist-packages/torch/functional.py:75:0
  %8 : Long(2, strides=[1], device=cpu) = onnx::Shape(%5) # /usr/local/lib/python3.7/dist-packages/torch/functional.py:75:0
  %9 : Float(2, 1, strides=[1, 1], requires_grad=0, device=cpu) = onnx::Expand(%1, %8) # /usr/local/lib/python3.7/dist-packages/torch/functional.py:75:0
  %10 : Long(requires_grad=0, device=cpu) = onnx::Constant[value={2}]() # <ipython-input-5-6eed90abf6ca>:3:0
  return (%10, %7, %9)
```

Maybe this got fixed at some point (maybe in https://github.com/pytorch/pytorch/pull/33789)?

**Comment by garymm** [comment]
Awaiting responses from @metalprogrammer to my comments [here](https://github.com/pytorch/pytorch/pull/66263#discussion_r741366575).

If there's no response in a few weeks I'll close this PR.

**Pull request closed without merging**